### PR TITLE
Add Experiment: Shared links on nav between docs <> devhub

### PR DESCRIPTION
**Pull request by teesloane**
Ticket: [CIRCLE-37883](https://circleci.atlassian.net/browse/CIRCLE-37883)

## Changes

Implement an experimental version of the navbar that is shared with the developer hub (Devhub [PR here](https://github.com/circleci/web-ui-developer-hub/pull/287))

Corresponding ticket for the Devhub side can be found [here](https://circleci.atlassian.net/browse/CIRCLE-37884).

The hope is that this will help users find images and orbs more easily, and make a more integrated user experience.

## Before

Screen Shot 2021-09-20 at 3 22 57 PM

## After
Screen Shot 2021-09-20 at 3 22 49 PM

### CODE REVIEW

**Review by tshen815 — APPROVED** [review]
both docs and 2.0 look good, nice work!

File: src-js/experiments/DevHubDocsNav.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  window.OptimizelyClient.getVariationName({
+    experimentKey: 'dd_merge_devhub_docs_experiment_test',
+    groupExperimentName: 'q3_fy22_docs_disco_experiment_group_test'
+  }).then(variation => {
+    if (variation === "treatment") {
+      $(".dd-global-nav--links").show();
+      $(".dd-devhub-docs-nav-link").click(function(event) {
+        event.preventDefault();
+        window.AnalyticsClient.trackAction('docs-devhub-nav-merge-link-clicked', {link: this.href})
+        window.location = this.href
```
Comment (rsylvian) [inline src-js/experiments/DevHubDocsNav.js:13]: curious why do we need `event.preventDefault();` and `window.location = this.href` ?

File: jekyll/_includes/global-nav.html [hunk 2]
```diff
@@ -59,5 +59,12 @@
         </div>
       </nav>
     </div>
+
+    <!-- Experiment:   // https://app.optimizely.com/v2/projects/16812830475/experiments/20616121296/variations -->
+    <div class="dd-global-nav--links" style="display: none">
```
Comment (rsylvian) [inline jekyll/_includes/global-nav.html:64]: Any reason why you used `--`?

File: jekyll/_includes/global-nav.html [hunk 3]
```diff
@@ -59,5 +59,12 @@
...
+      <a class="link home dd-devhub-docs-nav-link" href="/docs/2.0">Home</a>
+      <a class="link dd-devhub-docs-nav-link" href="https://circleci.com/developer/images">Images</a>
+      <a class="link dd-devhub-docs-nav-link" href="https://circleci.com/developer/orbs">Orbs</a>
```
Comment (rsylvian) [inline jekyll/_includes/global-nav.html:67]: Doesn't seems like you use the `dd-devhub-docs-nav-link` class for styling so I would suggest removing it to not clutter the page unnecessarily. When it comes to your event tracking you could just do  `$(".dd-global-nav--links a")`

**Review by rsylvian — CHANGES_REQUESTED** [review]

File: jekyll/_sass/_global-nav.scss [hunk 4]
```diff
@@ -251,7 +251,8 @@ $padding-large-vertical: 11px;
 .global-nav--search-bar {
   position: absolute;
   left: $leftbar-width;
-  top: 50%;
+  // top: 50%;
```
Comment (rsylvian) [inline jekyll/_sass/_global-nav.scss:254]:
```suggestion
```

File: src-js/experiments/DevHubDocsNav.js (hunk 1 above)
Comment (teesloane) [reply src-js/experiments/DevHubDocsNav.js:13]: I made the assumption that the <a> was navigating away from the page before the tracking invocation could execute. I will double check this is the case, though.

File: jekyll/_includes/global-nav.html (hunk 2 above)
Comment (teesloane) [reply jekyll/_includes/global-nav.html:64]: The styling methodology is a bit scattered, but it somewhat follows a bem-esque architecture.

File: src-js/experiments/DevHubDocsNav.js (hunk 1 above)
Comment (teesloane) [reply src-js/experiments/DevHubDocsNav.js]: Update - this was a bad assumption - it was unnecessary to preventDefault. I have updated this.

File: jekyll/_includes/global-nav.html [hunk 5]
```diff
@@ -54,10 +56,17 @@
...
           </a>
           <a href="https://app.circleci.com/dashboard" class="btn-primary dashboard-link">
-            Go to app
+            Go to Application
           </a>
         </div>
       </nav>
     </div>
+
+    <!-- Experiment:   // https://app.optimizely.com/v2/projects/16812830475/experiments/20616121296/variations -->
```
Comment (rsylvian) [inline jekyll/_includes/global-nav.html:65]:
```suggestion
    <!-- Experiment: https://app.optimizely.com/v2/projects/16812830475/experiments/20616121296/variations -->
```

File: jekyll/_sass/_global-nav.scss [hunk 6]
```diff
@@ -331,18 +335,33 @@ body.global-nav-open .global-nav--search-button {
...
+    border: none;
+    transition: background-color 300ms;
+    flex: auto 0 0;
+    color: #ffffff;
+    background-color: #049b4a;
+    width: 7rem;
 
     @media (max-width: $screen-xs-max) {
-      width: 100%;
+      // width: 100%;
```
Comment (rsylvian) [inline jekyll/_sass/_global-nav.scss:357]:
```suggestion
```

File: jekyll/_sass/_global-nav.scss [hunk 7]
```diff
@@ -440,3 +459,77 @@ a.nav-skip-button {
...
+      font-family: 'Roboto','Helvetica Neue',Helvetica,Arial,sans-serif;
+      font-size: 16px;
+      font-style: normal;
+      font-weight: 500;
+      line-height: 19px;
+      margin-right: 40px;
+      padding-bottom: 10px;
+      padding-top: 4px;
+      // padding: 3px 0px 19px 0;
+      // padding-top: 0px;
```
Comment (rsylvian) [inline jekyll/_sass/_global-nav.scss:487]:
```suggestion
```

File: jekyll/_sass/_global-nav.scss [hunk 8]
```diff
@@ -440,3 +459,77 @@ a.nav-skip-button {
...
+  .global-nav--lockup-title {
+    font-size: 25.2px;
+    margin-top: 2px;
+    margin-left: 8px;
+  }
+
+
+  .dd-word-mark-width {
+    // width: 125px;
+  }
```
Comment (rsylvian) [inline jekyll/_sass/_global-nav.scss:533]:
```suggestion
```

File: jekyll/_includes/utility-icons/external.svg [hunk 9]
```diff
@@ -1 +1,7 @@
...
\ No newline at end of file
+{% if include.size %}
+  {% assign size = include.size %}
+{% else %}
+  {% assign size = 12 %}
+{% endif %}
+
+<svg viewBox="0 0 {{size}} {{size}}" fill="currentColor" xmlns="http://www.w3.org/2000/svg" class="css-bleycz"><path viewBox="0 0 {{size}} {{size}}" fill-rule="evenodd" d="M17,19 L5,19 L5,7 L10,7 L10,5.00083798 L4.44085883,5.00083798 C4.05495161,4.9876138 3.68075435,5.13139282 3.4077085,5.39780868 C3.13466265,5.66422454 2.98730555,6.02933568 3.00085883,6.40587245 L3.00085883,19.5941275 C2.98730555,19.9706643 3.13466265,20.3357755 3.4077085,20.6021913 C3.68075435,20.8686072 4.05495161,21.0123862 4.44085883,20.999162 L17.56,20.999162 C18.35529,20.999162 19,20.3701067 19,19.5941275 L19,14 L17,14 L17,19 Z M17.5857864,5 L14,5 C13.4477153,5 13,4.55228475 13,4 C13,3.44771525 13.4477153,3 14,3 L20,3 C20.5522847,3 21,3.44771525 21,4 L21,10 C21,10.5522847 20.5522847,11 20,11 C19.4477153,11 19,10.5522847 19,10 L19,6.41421356 L14.8786797,10.5355339 C14.4881554,10.9260582 13.8549904,10.9260582 13.4644661,10.5355339 C13.0739418,10.1450096 13.0739418,9.51184464 13.4644661,9.12132034 L17.5857864,5 Z"></path></svg>
```
Comment (rsylvian) [inline jekyll/_includes/utility-icons/external.svg:7]: This icon flashes a different size when the page load:

Kapture 2021-10-01 at 19 26 38

Can you please fix this before our pairing? 😊

**Review by rsylvian — CHANGES_REQUESTED** [review]
Left couple of comments, nothing major. I also deployed your branch to a preview branch where you can see your work live: http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/test-preview/?force-all

**Comment by ccgeller** [comment]
Could the description be updated with more detail?
For example in the following PR:
https://github.com/circleci/web-ui-developer-hub/pull/287

File: jekyll/_includes/utility-icons/external.svg (hunk 9 above)
Comment (teesloane) [reply jekyll/_includes/utility-icons/external.svg:7]: Done!

**Comment by teesloane** [comment]
> Could the description be updated with more detail? For example in the following PR: [circleci/web-ui-developer-hub#287](https://github.com/circleci/web-ui-developer-hub/pull/287)

Good call - will do.

**Comment by nramabad** [comment]
the padding appears to be 15px on the icon & nav links as opposed to 29px on the mockup 

see more: https://circleci.slack.com/archives/C021NCSEM44/p1633368430227700?thread_ts=1633363656.222600&cid=C021NCSEM44

File: jekyll/_sass/_global-nav.scss [hunk 10]
```diff
@@ -331,18 +345,32 @@ body.global-nav-open .global-nav--search-button {
...
-    padding: $padding-large-vertical 30px;
-    width: auto;
-    height: auto;
-    margin-left: auto;
-    margin-right: auto;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    align-self: center;
+    padding: 12px 16px 12px 16px;
```
Comment (tshen815) [inline jekyll/_sass/_global-nav.scss:352]: absolute nit here but what about `padding: 12px 16px`

Comment (teesloane) [reply jekyll/_sass/_global-nav.scss:352]: True!

**Review by ccgeller — APPROVED** [review]
Works for me!

**Review by rsylvian — APPROVED** [review]

**Review by rsylvian — APPROVED** [review]
✅  All good! Paired with @teesloane & @tshen815 on this!

**Pull request merged**